index.d.ts

```diff
@@ -300,7 +300,8 @@ declare module "leap-core" {
   }
 
   class ExtendedWeb3 extends Web3 {
-    public getUnspent(address: string, cb?: Callback<Array<Unspent>>): Promise<Array<Unspent>>;
+    public getUnspent(address: string, color: number, cb?: Callback<Array<Unspent>>): Promise<Array<Unspent>>;
+    public getUnspentByAddress(address: string, cb?: Callback<Array<Unspent>>): Promise<Array<Unspent>>;
```

Comment:
TypeScript compiler not so stupid :-) You can have both declarations with one name:

```ts
public getUnspent(address: string, cb?: Callback<Array<Unspent>>): Promise<Array<Unspent>>;
public getUnspent(address: string, color: number, cb?: Callback<Array<Unspent>>): Promise<Array<Unspent>>;
```

Both will be valid

Reply:
but what if I use leap-core from plain js ?

Reply:
not sure it will be possible to have just one `getUnspent` method accepting variable number of parameters, because of how they are declared with Web3 code: [this](https://github.com/leapdao/leap-core/pull/110/files#diff-27a1c86133de4310373decfdad200a91R91) won't allow 1 argument, I think.

Reply:
As a workaround, we can create two methods — `getUnspentByAddrColor` and `getUnspentByAddr`. Then add new method `getUnspent`:

```es6
web3Instance.getUnspent = (...params) => {
  const last = params[params.length - 1];
  const hasCb = typeof last === 'function';
  if (params.length === (hasCb ? 2 : 1)) {
    return web3Instance.getUnspentByAddr(...params);
  }

  if (params.length === (hasCb ? 3 : 2)) {
     return web3Instance.getUnspentByAddrColor(...params);
  }
}
```

But maybe too much effort for such thing. Two methods fine too, just need to bump major version in this case